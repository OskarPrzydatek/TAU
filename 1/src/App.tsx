import LoginForm from './components/LoginForm/LoginForm';

export default function App() {
  const onLoginForm = (email: string, password: string) => {
    // Form data usage simulation
    console.log({ email: email, password: password });
  };

  return (
    <div className="App">
      <LoginForm onLoginSubmit={onLoginForm} />
    </div>
  );
}
