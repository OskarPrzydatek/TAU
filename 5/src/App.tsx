import React from 'react';
import calculatorReducer from './calculatorReducer';

import Button from './components/Button';
import NumericInput from './components/NumericInput';
import ResultView from './components/ResultView';

import { Operation } from './consts/operation.enum';
import { calculatorInitState } from './consts/calculatorInitState';

export default function App() {
  const [state, dispatch] = React.useReducer(
    calculatorReducer,
    calculatorInitState,
    (init) => init
  );

  const alphaInputRef = React.useRef<HTMLInputElement>(null);
  const betaInputRef = React.useRef<HTMLInputElement>(null);

  const handleOperation = (operation: Operation) => {
    const alpha = alphaInputRef.current?.value;
    const beta = betaInputRef.current?.value;

    dispatch({
      type: operation,
      alpha: Number(alpha),
      beta: Number(beta),
    });
  };

  const add = () => handleOperation(Operation.ADD);

  const sub = () => handleOperation(Operation.SUB);

  const mul = () => handleOperation(Operation.MUL);

  const div = () => handleOperation(Operation.DIV);

  return (
    <div className="App">
      <h1>Simple Calculator</h1>
      <ResultView
        resultTestID="calculator-result"
        alpha={state.alpha}
        beta={state.beta}
        operation={state.operationSign}
        result={state.result}
      />
      <NumericInput
        dataTestID="alpha-input"
        ref={alphaInputRef}
        label="Alpha: "
      />
      <NumericInput dataTestID="beta-input" ref={betaInputRef} label="Beta: " />
      <Button dataTestID="add-button" label={Operation.ADD} onClick={add} />
      <Button dataTestID="sub-button" label={Operation.SUB} onClick={sub} />
      <Button dataTestID="mul-button" label={Operation.MUL} onClick={mul} />
      <Button dataTestID="div-button" label={Operation.DIV} onClick={div} />
    </div>
  );
}
