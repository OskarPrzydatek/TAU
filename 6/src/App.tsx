import React from "react";
import { useDispatch, useSelector } from "react-redux";

import { Board, Field, Game, Modal, Row } from "./components";
import { Button } from "./components/Button.component";
import { HomePage } from "./pages";

import {
  move,
  newGame,
  selectBoard,
  selectPlayerDestination,
  selectPlayerPosition,
} from "./store/game/game.slice";

import "./styles/styles.css";

function App() {
  const dispatch = useDispatch();
  const board = useSelector(selectBoard);
  const playerPosition = useSelector(selectPlayerPosition);
  const playerDestination = useSelector(selectPlayerDestination);

  const gameWinCondition = playerDestination === playerPosition;

  const onClickPlayAgain = () => dispatch(newGame());

  console.log(board)

  React.useEffect(() => {
    const keyboardListner = (event: KeyboardEvent) => {
      dispatch(move(event.key));
    };

    document.addEventListener("keydown", keyboardListner);

    return () => {
      document.removeEventListener("keydown", keyboardListner);
    };
  }, [dispatch]);

  return (
    <HomePage
      board={board}
      playerPosition={playerPosition}
      gameWinCondition={gameWinCondition}
      onClickPlayAgain={onClickPlayAgain}
    />
  );
}

export default App;
