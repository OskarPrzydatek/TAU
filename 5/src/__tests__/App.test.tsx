import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import App from '../App';

describe('App', () => {
  test('Component Snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure add works correctly', () => {
    render(<App />);
    fireEvent.change(screen.getByTestId('alpha-input'), {
      target: { value: '3' },
    });
    fireEvent.change(screen.getByTestId('beta-input'), {
      target: { value: '3' },
    });
    fireEvent.click(screen.getByTestId('add-button'));
    expect(screen.getByTestId('calculator-result')).toHaveTextContent(
      '3 + 3 = 6'
    );
  });

  test('ensure sub works correctly', () => {
    render(<App />);
    fireEvent.change(screen.getByTestId('alpha-input'), {
      target: { value: '10' },
    });
    fireEvent.change(screen.getByTestId('beta-input'), {
      target: { value: '5' },
    });
    fireEvent.click(screen.getByTestId('sub-button'));
    expect(screen.getByTestId('calculator-result')).toHaveTextContent(
      '10 - 5 = 5'
    );
  });

  test('ensure mul works correctly', () => {
    render(<App />);
    fireEvent.change(screen.getByTestId('alpha-input'), {
      target: { value: '2' },
    });
    fireEvent.change(screen.getByTestId('beta-input'), {
      target: { value: '2' },
    });
    fireEvent.click(screen.getByTestId('mul-button'));
    expect(screen.getByTestId('calculator-result')).toHaveTextContent(
      '2 * 2 = 4'
    );
  });

  test('ensure div works correctly', () => {
    render(<App />);
    fireEvent.change(screen.getByTestId('alpha-input'), {
      target: { value: '2' },
    });
    fireEvent.change(screen.getByTestId('beta-input'), {
      target: { value: '2' },
    });
    fireEvent.click(screen.getByTestId('div-button'));
    expect(screen.getByTestId('calculator-result')).toHaveTextContent(
      '2 / 2 = 1'
    );
  });
});
