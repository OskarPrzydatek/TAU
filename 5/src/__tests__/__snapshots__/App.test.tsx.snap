// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Snapshot 1`] = `
<div
  className="App"
>
  <h1>
    Simple Calculator
  </h1>
  <h2>
    Put numbers to inputs and choose operation, then You'll see the result :D
  </h2>
  <label>
    <span>
      Alpha: 
    </span>
    <input
      data-testid="alpha-input"
      onKeyDown={[Function]}
      type="number"
    />
  </label>
  <label>
    <span>
      Beta: 
    </span>
    <input
      data-testid="beta-input"
      onKeyDown={[Function]}
      type="number"
    />
  </label>
  <button
    data-testid="add-button"
    onClick={[Function]}
  >
    +
  </button>
  <button
    data-testid="sub-button"
    onClick={[Function]}
  >
    -
  </button>
  <button
    data-testid="mul-button"
    onClick={[Function]}
  >
    *
  </button>
  <button
    data-testid="div-button"
    onClick={[Function]}
  >
    /
  </button>
</div>
`;
